app/main/views/g_cloud.py

```diff
@@ -603,10 +603,39 @@ def which_service_won_contract(framework_framework, project_id):
 
 @direct_award.route(
     '/<string:framework_framework>/projects/<int:project_id>/why-didnt-you-award-contract',
-    methods=['GET']
+    methods=['GET', 'POST']
 )
-def why_didnt_you_award_contract(framework_framework, project_id):
-    abort(404)
+def why_did_you_not_award_the_contract(framework_framework, project_id):
+    all_frameworks = data_api_client.find_frameworks().get('frameworks')
+    framework = framework_helpers.get_latest_live_framework(
+        all_frameworks, framework_framework)
+
+    project = data_api_client.get_direct_award_project(project_id=project_id)['project']
+
+    if not is_direct_award_project_accessible(project, current_user.id):
+        abort(404)
+
+    if not project['lockedAt']:
+        abort(400)
+
+    form = WhyDidYouNotAwardForm()
+
+    if request.method == "POST" and form.validate_on_submit():
+        flash('You’ve updated ‘' + project['name'] + '’')
+        return redirect(url_for('.view_project', framework_framework=framework_framework, project_id=project['id']))
+
+    errors = [{
+        'input_name': input_name,
+        'question': getattr(form, input_name).label,
+    } for input_name in form.errors.keys()]
```

Comment:
I think @lfdebrux is planning to move this common thing into a function, so would be good if you can figure out how to do use the same code :-)

Reply:
It's now in dmutils 37.0.0 as `forms.get_errors_from_wtform`. Albeit pulling that utils update in does recommend some larger changes to the frontend, so it might be a pain pulling that in. I might suggest leaving it as-is here, until I can get around to pulling it in. I'm halfway through the work on the buyer-fe.